pallets/pool-system/src/impls.rs

```diff
@@ -367,6 +372,60 @@ impl<T: Config> InvestmentAccountant<T::AccountId> for Pallet<T> {
 	}
 }
 
+impl<T: Config> ChangeGuard for Pallet<T> {
+	type Change = T::RuntimeChange;
+	type ChangeId = T::Hash;
+	type PoolId = T::PoolId;
+
+	fn note(pool_id: Self::PoolId, change: Self::Change) -> Result<Self::ChangeId, DispatchError> {
+		let registered_change = (Self::now(), change.clone());
+		let change_id: Self::ChangeId = T::Hashing::hash(&registered_change.encode());
+		Changes::<T>::insert(pool_id, change_id, registered_change);
+
+		Self::deposit_event(Event::ProposedChange {
+			pool_id,
+			change_id,
+			change,
+		});
+
+		Ok(change_id)
+	}
+
+	fn released(
+		pool_id: Self::PoolId,
+		change_id: Self::ChangeId,
+	) -> Result<Self::Change, DispatchError> {
+		let (submitted_time, change) =
+			Changes::<T>::get(pool_id, change_id).ok_or(Error::<T>::ChangeNotFound)?;
+
+		let pool_change: PoolChangeProposal = change.clone().into();
+
+		let mut allowed = true;
+		for requirement in pool_change.requirements() {
+			allowed &= match requirement {
+				Requirement::NextEpoch => {
+					match Pool::<T>::get(pool_id) {
+						Some(details) => submitted_time < details.epoch.last_closed,
```

Comment:
Not sure if we should use `last_closed` or `last_executed` here

Reply:
I would opt for `last_executed` as we can have cash flows in between. Although, those should not change the the execution/solution of previous. But I think this will change when we change the rebalancing.

Reply:
Ok, `last_executed` makes more sense!

Reply:
Looking into it, I thought the `last_executed` was the `Moment`, but it's an `EpochId` so I can not compare it with the `submitted_time` I'm already handle. Solutions:
1. Keep it by now as it is, using the `last_closed`.
2. Store the `EpochId` when the event was noted, to check it later against `last_executed`.

Reply:
I think we can go with 1., if this is ensured for all changes: https://github.com/centrifuge/centrifuge-chain/pull/1396#discussion_r1229242865

Reply:
I think `last_closed` is fine actually. This requirement just needs to ensure that at least the epoch is inbetween noting the change and releasing it. This is true once `last_closed > submitted_time`. The time inbetween closing and execution does not matter if changes cannot be executed during the submission period anyways, which is the other default requirement we are adding.

Reply:
Yeah, composing this rule with the default rule we are safe 👍🏻